x/staking/client/cli/tx.go

```diff
@@ -411,7 +412,8 @@ func newBuildCreateValidatorMsg(clientCtx client.Context, txf tx.Factory, fs *fl
 // this is anticipated to be used with the gen-tx
 func CreateValidatorMsgFlagSet(ipDefault string) (fs *flag.FlagSet, defaultsDesc string) {
 	fsCreateValidator := flag.NewFlagSet("", flag.ContinueOnError)
-	fsCreateValidator.String(FlagIP, ipDefault, "The node's public IP")
+	fsCreateValidator.String(FlagIP, ipDefault, "The node's public P2P IP")
+	fsCreateValidator.Uint(FlagP2PPort, 26656, "The node's public P2P port")
```

Comment:
This default value should be the port value from `.p2p.laddr` specified in `config.toml`. Don't you agree?

Reply:
No, it should be the default of what Tendermint provides, which is 26656.

Reply:
Also, you cannot cleanly statically link a default value to a dynamic value read from a configuration file.

Reply:
I think, if the `external_address` is set to a valid IP address in `config.toml`, `gentx` command should take that as the default value. The same goes for `laddr` port.

Probably I did not explain the situation clearly. I stumbled upon this problem while deploying a local testnet. After calling `simd init` for multiple validators, I changed the `laddr` ports in each `config.toml` so that each validator has a unique port. But when I call `gentx` command for each validator, all the generated transactions hardcodes `26656` - even though I set different values in `config.toml.`

In real life, this issue can come up if a genesis validator decides to use a non-default p2p port.

Reply:
I know, I understand the context of the problem and I think having the default as 26656 for the `p2p-port` is perfectly sufficient -- if you don't expect to use that port, then use the flag, that's what it's there for.

Reply:
Hm. This makes sense - the port flag should be enough. I agree with you now.

Thanks for the PR :slightly_smiling_face: